src/derivable/mixins/pluck.spec.ts

```diff
@@ -44,14 +45,14 @@ export function testPluck(factory: Factory) {
             expect(value1$.get()).to.equal('value2');
         });
 
-        context('(lensed with constant key)', () => {
+        if (isSettable) {
             class MyClass { constructor(public key: string) { } }
-            const value$ = factory(new MyClass('value'));
+            let value$: SettableDerivable<MyClass>;
+            beforeEach('setup the base', () => { value$ = assertSettable(factory(new MyClass('value'))); });
 
-            if (isSettableDerivable(value$)) {
-                beforeEach('reset the atom', () => value$.set(new MyClass('value')));
-
-                const plucked$ = value$.pluck('key');
+            context('(lensed with constant key)', () => {
+                let plucked$: SettableDerivable<string>;
+                beforeEach('setup the derivation', () => { plucked$ = value$.pluck('key'); });
 
                 it('should produce a SettableDerivable if the base was also a SettableDerivable', () => {
```

Comment:
For 'spec' purposes, maybe also add a test that iff the input is `DerivableAtom`, it will produce a `DerivableAtom`.